frontend/packages/topology/src/components/page/DroppableTopologyComponent.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { NativeTypes } from 'react-dnd-html5-backend';
+import { DropTarget, DropTargetConnector } from 'react-dnd';
+import { DropTargetMonitor } from 'react-dnd/lib/interfaces';
+import { Model } from '@patternfly/react-topology';
+import withDragDropContext from '@console/internal/components/utils/drag-drop-context';
+import {
+  FileUploadContextType,
+  FileUploadContext,
+} from '@console/app/src/components/file-upload/file-upload-context';
+import { ConnectedTopologyView, TopologyViewProps } from './TopologyView';
+import { TopologyViewType } from '../../topology-types';
+
+const boxTarget = {
+  drop(props, monitor) {
+    if (props.onDrop && monitor.isOver()) {
+      props.onDrop(props, monitor);
+    }
+  },
+};
+
+const DroppableTopology = DropTarget(
+  NativeTypes.FILE,
+  boxTarget,
+  (connectObj: DropTargetConnector, monitor: DropTargetMonitor, props: TopologyViewProps) => {
+    return {
+      connectDropTarget: connectObj.dropTarget(),
+      isOver: monitor.isOver(),
+      canDrop: monitor.canDrop() && props.fileExtensions.length > 0,
+    };
+  },
+)(ConnectedTopologyView);
+
+export const DroppableTopologyComponent = withDragDropContext<DroppableTopologyComponentProps>(
+  (props) => {
+    const { setFileUpload, extensions } = React.useContext<FileUploadContextType>(
+      FileUploadContext,
+    );
+
+    const handleFileDrop = (item, monitor: DropTargetMonitor) => {
+      if (!monitor) {
+        return;
+      }
+      const [file] = monitor.getItem().files;
+      if (!file) {
+        return;
+      }
+      setFileUpload(file);
+    };
+
+    return <DroppableTopology {...props} onDrop={handleFileDrop} fileExtensions={extensions} />;
```

Comment:
just a suggestion: DroppableTopology component doesn't need to know about the fileExtensions, just add a prop canDrop
```suggestion
    return <DroppableTopology {...props} onDrop={handleFileDrop} canDrop={extensions.length > 0} />;
```